codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/Route53Customizations.java

```diff
@@ -0,0 +1,158 @@
+                        .operationPredicate(Route53Customizations::supportsHostedZoneIDValue)
+                        .registerMiddleware(MiddlewareRegistrar.builder()
+                                .resolvedFunction(SymbolUtils.createValueSymbolBuilder(URL_SANITIZE_ADDER).build())
+                                .build())
+                        .build()
+        );
+    }
+
+
+    public void writeAdditionalFiles(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            GoDelegator goDelegator
+    ) {
+        if (!isRoute53Service(model, settings.getService(model))) {
+            return;
+        }
+
+        ServiceShape service = settings.getService(model);
+        goDelegator.useShapeWriter(service, this::writeMiddlewareHelper);
+
+        goDelegator.useShapeWriter(service, writer -> {
+                writeHostedZoneIDInputSanitizer(writer, model, symbolProvider, service);
+            });
+    }
+
+    private void writeMiddlewareHelper(GoWriter writer) {
+        writer.openBlock("func $L(stack *middleware.Stack) {", "}", URL_SANITIZE_ADDER, () -> {
+            writer.write("$T(stack, $T{SanitizeHostedZoneIDInput: $L})",
+                    SymbolUtils.createValueSymbolBuilder(URL_SANITIZE_INTERNAL_ADDER,
+                            AwsCustomGoDependency.ROUTE53_CUSTOMIZATION).build(),
+                    SymbolUtils.createValueSymbolBuilder(URL_SANITIZE_INTERNAL_ADDER + "Options",
+                            AwsCustomGoDependency.ROUTE53_CUSTOMIZATION).build(),
+                    SANITIZE_HOSTED_ZONE_ID_INPUT
+            );
+        });
+        writer.insertTrailingNewline();
+    }
+
+    private void writeHostedZoneIDInputSanitizer(
+            GoWriter writer,
+            Model model,
+            SymbolProvider symbolProvider,
+            ServiceShape service
+    ) {
+
+        writer.writeDocs("Check for and split apart Route53 resource IDs, setting only the last piece. " +
+                "This allows the output of one operation e.g. foo/1234 to be used as input in another operation " +
+                "(e.g. it expects just '1234')");
+
+        writer.openBlock("func sanitizeHostedZoneIDInput(input interface{}) error {", "}", () -> {
+            writer.openBlock("switch i:= input.(type) {", "}", () -> {
+                service.getAllOperations().forEach((operationId)-> {
+                    OperationShape operation = model.expectShape(operationId, OperationShape.class);
+                    StructureShape input = model.expectShape(operation.getInput().get(), StructureShape.class);
+                    List<MemberShape> hostedZoneIDMembers = input.getAllMembers().values().stream()
+                            .filter(m -> m.getTarget().getName().equalsIgnoreCase("ResourceId")
+                                    || m.getTarget().getName().equalsIgnoreCase("DelegationSetId"))
+                            .collect(Collectors.toList());
```

Comment:
Should this customization target all `ResourceId` and `DelegationSetId` shapes, usages or only those usages that target the URL Path httpbinding?  The v1 SDK's customization would of only impacted API operation with the ID targeted the URL path, are there any cases where the resource ID targeted the query string or request body?

Reply:
I followed the boto customization which targets all `ResourceId` and `DelegationSetId` shapes, irrespective of usage.

https://github.com/boto/botocore/blob/develop/botocore/handlers.py#L540-L558